editor/export/editor_export_platform.cpp

```diff
@@ -831,29 +831,34 @@ String EditorExportPlatform::_get_script_encryption_key(const Ref<EditorExportPr
 	return p_preset->get_script_encryption_key().to_lower();
 }
 
-Vector<String> EditorExportPlatform::get_forced_export_files() {
+Vector<String> EditorExportPlatform::get_forced_export_files(bool p_main_pack) {
 	Vector<String> files;
 
-	files.push_back(ProjectSettings::get_singleton()->get_global_class_list_path());
+	// these files are only needed if we are exporting a main pack file
+	if (p_main_pack) {
+		String icon = GLOBAL_GET("application/config/icon");
+		if (!icon.is_empty() && FileAccess::exists(icon)) {
+			files.push_back(icon);
+		}
 
-	String icon = GLOBAL_GET("application/config/icon");
-	String splash = GLOBAL_GET("application/boot_splash/image");
-	if (!icon.is_empty() && FileAccess::exists(icon)) {
-		files.push_back(icon);
-	}
-	if (!splash.is_empty() && FileAccess::exists(splash) && icon != splash) {
-		files.push_back(splash);
+		String splash = GLOBAL_GET("application/boot_splash/image");
+		if (!splash.is_empty() && FileAccess::exists(splash) && icon != splash) {
```

Comment:
```suggestion
		if (!splash.is_empty() && icon != splash && FileAccess::exists(splash)) {
```
Put cheaper condition before more expensive ones